addon/components/modal-dialog.js

```diff
@@ -51,7 +51,16 @@ export default Ember.Component.extend({
     }
   }),
 
+  focusOnOpen() {
+    let focusSelector = this.get('focusSelector');
+    if (this.get('focusSelector')) {
+      Ember.$(focusSelector).focus();
```

Comment:
Seems like this should be scoped to the modal contents instead of matching across the whole document. I'm not sure if `this.$(focusSelector)` will work due to tether/wormhole behaviors, but it may be ok.